crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -177,21 +182,51 @@ impl ObjectsSnapshotProcessor {
                         .await?
                         .unwrap_or_default();
 
-                    // We update the snapshot table when it falls behind the rest of the indexer by more than the min_lag.
+                    // We update the snapshot table when it falls behind the rest of the indexer by
+                    // more than the min_lag. When `latest_indexer_cp = start_cp +
+                    // config.snapshot_min_lag`, we have not actually indexed `start_cp` yet, hence
+                    // why the condition is `>=`.
                     while latest_indexer_cp >= start_cp + config.snapshot_min_lag as u64 {
-                        // Stream the next object changes to be committed to the store.
-                        if let Some(object_changes_batch) = stream.next().await {
-                            let first_checkpoint_seq = object_changes_batch.first().as_ref().unwrap().checkpoint_sequence_number;
-                            let last_checkpoint_seq = object_changes_batch.last().as_ref().unwrap().checkpoint_sequence_number;
+                        // The maximum checkpoint sequence number that can be committed to the
+                        // `objects_snapshot` table.
+                        let max_allowed_cp = latest_indexer_cp - config.snapshot_min_lag as u64;
```

Comment:
(nit: a SAFETY comment is worthwhile here to explain why this subtraction is safe -- because of the `while` condition, and presumably because `latest_indexer_cp` has to be greater than or equal to `stat_cp`).